config.toml.sample

```diff
@@ -0,0 +1,12 @@
+[github]
+# A GitHub personal access token.
+access_token = "<token>"
+
+# A GitHub oauth application for this instance of homu:
+app_client_id = "<id>"
+app_client_secret = "<secret>"
```

Comment:
Homu used GitHub OAuth to authenticate, but is there a reason why we're not using a GitHub App? That provides granular permissions, configures webhooks and handles webhook secrets automatically for us.

With that, environment variables, and a `rust-bors.toml` at the root of each repository (similar to `triagebot.toml`) there should be no need to have a configuration file at all.

Reply:
I *think* that @Mark-Simulacrum mentioned something in the sense that it would be nice if we could manage the deployment ourselves (I assume that GitHub apps run on GH infrastructure, but I may be wrong, I have never used them).

Reply:
GitHub Apps are just a different way to authenticate with the API, it'd still be hosted by us. The main differences for us would be:

* Rather than setting up webhooks in each repository, there is a single webhook for the app that gets events for all the repositories the app is installed on.
* Permissions are more granular than personal access tokens.
* Enabling bors on a repository would just be installing the app on the new repository too, rather than having to add the bors user to the list of users with write access and configuring an ad-hoc webhook.

See https://docs.github.com/en/developers/apps/getting-started-with-apps/about-apps.